implemented a filter for auctions and bids lists

matthias-luger opened the pull request:
fixes #960

matthias-luger committed:
implemented a filter for auctions and bids lists

Ekwav reviewed (changes requested):
* Only own auction should be filterable if no paid account (and note of this fact) is missing
* Wiki and hyauctions link should not be displayed 
![image](https://user-images.githubusercontent.com/16632490/198882598-23cea894-8869-4476-aed5-8b9b3006455b.png)
* I am not quite sure but when adding and removing filters the filtered results stay and aren't refreshed currently

matthias-luger committed:
hide options in player list filter

matthias-luger committed:
- fix loading issue
- only show filters if starter premium and above
- fix issue with wasAlreadyLoggedIn-Hook

matthias-luger committed:
added info how to enable filter

matthias-luger committed:
fix build error

Ekwav commented on pages/player/[uuid]/index.tsx:
```diff
@@ -160,6 +156,23 @@ function PlayerDetails(props: Props) {
+                            </>
+                        }
+                    />
+                ) : null}
```
please move this down below the buttons (where the filter would be) so users see it where they would search for the filter option

matthias-luger committed:
moved filter info

Ekwav commented on components/PlayerDetailsList/PlayerDetailsList.tsx:
```diff
@@ -316,6 +317,23 @@ function PlayerDetailsList(props: Props) {
+                            How to filter auctions/bids <HelpIcon style={{ color: '#007bff' }} />
+                        </p>
+                    }
+                    hoverPlacement="bottom"
+                    type="hover"
+                    tooltipContent={
+                        <>
+                            <p>Claim your account to filter your own auctions/bids.</p>
+                            <p>If you have starter premium or above you are able to use the filter for any player.</p>
+                        </>
+                    }
```
![bad centered](https://user-images.githubusercontent.com/16632490/201374138-ba6f51b5-c05a-46a9-8bee-1e619b31dd20.png)
should be below the information icon not in the center.  
Maybe add a redirection possibility to /premium as well

matthias-luger committed:
fix tooltip being wrongly centered

matthias-luger committed:
added premium link

matthias-luger committed:
Merge branch 'develop' into filter-auctions-and-bids
